cmd/show/show.go

```diff
@@ -0,0 +1,211 @@
+// Package bake contains the bootstrapping and tooling for the pizza bake
+// cobra command
+package show
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+// Options are the options for the pizza show command including user
+// defined configurations
+type Options struct {
+	// RepoName is the git repo name to get statistics
+	RepoName string
+
+	// Page is the page to be requested
+	Page int
+
+	// Limit is the number of records to be retrieved
+	Limit int
+
+	// Range is the number of days to take into account when retrieving statistics
+	Range int
+}
+
+const showLongDesc string = `WARNING: Proof of concept feature.
+
+The show command accepts a URL to a git repository and uses OpenSauced api
+to retrieve statistics of the repository and displayed them.`
+
+// NewBakeCommand returns a new cobra command for 'pizza bake'
+func NewShowCommand() *cobra.Command {
+	opts := &Options{}
+
+	cmd := &cobra.Command{
+		Use:   "show repository-name [flags]",
+		Short: "Uses OpenSauced api to get visual statistics of a repository",
+		Long:  showLongDesc,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errors.New("must specify the URL of a git repository to analyze")
+			}
+			opts.RepoName = args[0]
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(opts)
+		},
+	}
+
+	cmd.Flags().IntVarP(&opts.Range, "range", "r", 30, "The last N number of days to consider")
+	cmd.Flags().IntVarP(&opts.Limit, "limit", "l", 10, "The number of records to retrieve")
+	cmd.Flags().IntVarP(&opts.Page, "page", "p", 1, "The page number to retrieve")
+
+	return cmd
+}
+
+type Author struct {
+	Name string `json:"author_login"`
+}
+
+type Metadata struct {
+	Page            int  `json:"page"`
+	Limit           int  `json:"limit"`
+	ItemCount       int  `json:"itemCount"`
+	PageCount       int  `json:"pageCount"`
+	HasPreviousPage bool `json:"hasPreviousPage"`
+	HasNextPage     bool `json:"hasNextPage"`
+}
+
+type showRespose struct {
+	Contributors []Author `json:"data"`
+	Metadata     Metadata `json:"meta"`
+}
+
+type reposResponse struct {
+	ID                int       `json:"id"`
+	Size              int       `json:"size"`
+	Issues            int       `json:"issues"`
```

Comment:
Incredible work so far, you've been crushing it!!

I've been wondering if we should create some sort of generated Go package that makes working with the OpenSauced API easier.

This is good for now, but I worry about drift happening between these inline structs and API changes we make. cc @brandonroberts - I'll look at if there's anything possible with generating Go structs and APIs from swagger or NextJS definitions.

Reply:
@k1nho - The generated Go API has landed 👀 

https://github.com/open-sauced/go-api

Let's try using that and open issues in that repo with any sharp edges you might hit!